h2/stream.py

```diff
@@ -269,6 +282,10 @@ def recv_push_promise(self, previous_state):
         event.parent_stream_id = self.stream_id
         return [event]
 
+    def send_end_stream(self, previous_state):
+        if previous_state is StreamState.HALF_CLOSED_REMOTE:
```

Comment:
Same note here: it's better to have this function be called only in the appropriate state than to do this checking.